scripts/semconv/templates/registry/semantic_attributes.j2

```diff
@@ -0,0 +1,72 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Final
+
+{% set file_name = ctx.output + (ctx.root_namespace | kebab_case | replace('-', '_')) ~ "_attributes.py" -%}
+{{- template.set_file_name(file_name) -}}
+{%- import 'common.j2' as c %}
+
+{% set attributes = ctx.attributes | sort(attribute="name") | list %}
+{% set enum_attributes = attributes | selectattr("type", "mapping") | rejectattr("name", "in", ctx.excluded_attributes) | list %}
```

Comment:
There are multiple filters and tests for enums. 

Filters:
- `enum_type`: Filters a type to return the enum type or an error if the type is not an enum.
- `instantiated_type`: Filters a type to return the instantiated type. 

`instantiated_type` is more general as it can be applied to any attribute type and combined to `map_text` to produce the data type for the targeted PL.

Tests:
- `enum`: Tests if an attribute has an enum type.
- `simple_type`: Tests if a type is a simple type (i.e.: string | string[] | int | int[] | double | double[] | boolean | boolean[]).
- `template_type`: Tests if a type is a template type (i.e.: template[]).
- `enum_type`: Tests if a type is an enum type.